WordPress/openverse
# Update the external search modal and items

## Fixes



Fixes #460 by @fcoveram

## Description


This PR updates the external search items modal according to the updated designs.
- external sources use modal on all breakpoints (instead of selecting modal for small breakpoints and popover for larger ones). This allows us to use `VModal` to simplify the component, instead of handling the modal behavior manually.
- "centered" modal is full width, except for 24px of horizontal margins, on `xs`, and has a max-width set to 400px for breakpoints `sm` and above
- the sizes, paddings and text styles of the external sources buttons were updated

The `VModalContent` had a bug where the `hide` prop in the top-bar slot was not passed to the `VModal`, so it was impossible to close the modal using the close button. This PR fixes that, too.

## Designs
**Figma file:** [Metasearch improvement](https://www.figma.com/file/AsD7zOnUQxuLaUBiptTYAn/Metasearch-improvement?type=design&node-id=22-13656&mode=dev).

## Testing Instructions

Search for images or audio, scroll to the bottom and click on the "External sources" button. The modal should match the designs, should be closable, and should update sizes if you change the screen width. You should be able to open the external sources links.

## Checklist



[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
